Gems/ROS2/Code/Source/Frame/ROS2FrameComponent.cpp

```diff
@@ -203,49 +205,26 @@ namespace ROS2
         {
             parentNamespace = parentFrame->GetNamespace();
         }
-        return m_namespaceConfiguration.GetNamespace(parentNamespace);
+        return m_configuration.m_namespaceConfiguration.GetNamespace(parentNamespace);
     }
 
     AZ::Name ROS2FrameComponent::GetJointName() const
     {
-        return AZ::Name(ROS2Names::GetNamespacedName(GetNamespace(), m_jointNameString).c_str());
+        return AZ::Name(ROS2Names::GetNamespacedName(GetNamespace(), m_configuration.m_jointNameString).c_str());
     }
 
     void ROS2FrameComponent::SetJointName(const AZStd::string& jointNameString)
     {
-        m_jointNameString = jointNameString;
+        m_configuration.m_jointNameString = jointNameString;
     }
 
     void ROS2FrameComponent::Reflect(AZ::ReflectContext* context)
     {
-        NamespaceConfiguration::Reflect(context);
+        ROS2FrameConfiguration::Reflect(context);
         if (AZ::SerializeContext* serialize = azrtti_cast<AZ::SerializeContext*>(context))
         {
-            serialize->Class<ROS2FrameComponent, AZ::Component>()
-                ->Version(1)
-                ->Field("Namespace Configuration", &ROS2FrameComponent::m_namespaceConfiguration)
-                ->Field("Frame Name", &ROS2FrameComponent::m_frameName)
-                ->Field("Joint Name", &ROS2FrameComponent::m_jointNameString)
-                ->Field("Publish Transform", &ROS2FrameComponent::m_publishTransform);
-
-            if (AZ::EditContext* ec = serialize->GetEditContext())
-            {
-                ec->Class<ROS2FrameComponent>("ROS2 Frame", "[ROS2 Frame component]")
-                    ->ClassElement(AZ::Edit::ClassElements::EditorData, "")
-                    ->Attribute(AZ::Edit::Attributes::Category, "ROS2")
-                    ->Attribute(AZ::Edit::Attributes::AppearsInAddComponentMenu, AZ_CRC_CE("Game"))
-                    ->Attribute(AZ::Edit::Attributes::Icon, "Editor/Icons/Components/ROS2Frame.svg")
-                    ->Attribute(AZ::Edit::Attributes::ViewportIcon, "Editor/Icons/Components/Viewport/ROS2Frame.svg")
-                    ->DataElement(
-                        AZ::Edit::UIHandlers::Default,
-                        &ROS2FrameComponent::m_namespaceConfiguration,
-                        "Namespace Configuration",
-                        "Namespace Configuration")
-                    ->DataElement(AZ::Edit::UIHandlers::Default, &ROS2FrameComponent::m_frameName, "Frame Name", "Frame Name")
-                    ->DataElement(AZ::Edit::UIHandlers::Default, &ROS2FrameComponent::m_jointNameString, "Joint Name", "Joint Name")
-                    ->DataElement(
-                        AZ::Edit::UIHandlers::Default, &ROS2FrameComponent::m_publishTransform, "Publish Transform", "Publish Transform");
-            }
+            serialize->Class<ROS2FrameComponent, AZ::Component>()->Version(1)->Field(
+                "ROS2FrameConfiguration", &ROS2FrameComponent::m_configuration);
```

Comment:
Please add ShowOnlyChildren atrribute to adjust rendering in the Editor.

Reply:
This is not needed, as the ROS2FrameComponent is no longer visible in the Editor as it has been replaced by the ROS2FrameEditorComponent. The ROS2FrameComponent does not reflect anything in the edit context.

Reply:
The frame component shows only children in dpe
![image](https://github.com/o3de/o3de-extras/assets/130671280/99a71688-c287-4ec7-a347-fa5fdd012e22)